### ISSUE TITLE
[NAS Gateway] Kafka notification endpoints produce duplicate events

### ISSUE THREAD

**Issue by juv** [minio/minio#9875]
In MinIO `nas gateway` mode, there is a bug that ends up in duplicate Kafka messages for MinIO events such as PUT/DELETE. The error happens frequently, but is not 100% reproducible. 

## Expected Behavior

MinIO events such as PUT/DELETE are produced only once into the Kafka topic

## Current Behavior

MinIO events are occasionally produced multiple times into the Kafka topic and an error is written into the server log. 

```
API: SYSTEM()
Time: 09:07:18 UTC 06/19/2020
DeploymentID: b5ef63c2-2994-4559-a06e-c4307c18fbe1
Error: target.Send() failed with 'remove /tmp/minio/minio-kafka-bucket1/0229c718-f928-419a-9aea-1ef36fde1c74.event: no such file or directory'
       4: github.com/minio/minio@/cmd/logger/logonce.go:52:logger.(*logOnceType).logOnceIf()
       3: github.com/minio/minio@/cmd/logger/logonce.go:80:logger.LogOnceIf()
       2: github.com/minio/minio@/pkg/event/target/store.go:112:target.sendEvents.func1()
       1: github.com/minio/minio@/pkg/event/target/store.go:136:target.sendEvents()
```

## Steps to Reproduce (for bugs)

1. Enable MinIO in NAS gateway mode
2. Enable Kafka notification endpoint by setting the [environment variables ](https://docs.min.io/docs/minio-bucket-notification-guide.html#apache-kafka) for a bucket `bucket1`
3. Use `mc` or the MinIO web UI to put/delete items into `bucket1` multiple times -- multiple times is required to force this error. It is not happening 100% of the times.
4. Check Kafka topic for new (duplicate) messages and watch for the error above in the server log

## Context

We use the MinIO events to trigger processing of the files that get PUT into the buckets. When multiple messags for the same event get produced to Kafka, this will end up in (unnecessary) complexity in the consuming applications because they would need to build a queue and/or keep track of MinIO files that have already been processed.

## Your Environment

* Version used (`minio version`): RELEASE.2020-06-18T02-23-35Z
* Environment name and version (e.g. nginx 1.9.1): 
* Server type and version:
* Operating System and version (`uname -a`): Openshift 3.11
* Link to your project:

Here is a duplicate event, which was produced multiple times to my topic:
[minio-duplicate-event.txt](https://github.com/minio/minio/files/4803750/minio-duplicate-event.txt)

**Comment by juv** [comment]
If I set the `QueueDir` argument to `""`, I am unable to reproduce this issue - which means that it is working as expected. However, I am losing the replay capabilities that MinIO offers. 

When the `QueueDir` argument is not present, i.e. `""`, no queue gets created for the kafka target: https://github.com/minio/minio/blob/068ad33c9a1bf52b81597dcc271ab80390f9926b/pkg/event/target/kafka.go#L309-L317 

According to the error message above, something is going wrong when deleting an event file from the queue directory. Hope this helps.

Edit:
Maybe checking whether the file exists before trying to delete it from the queue directory might help here? https://github.com/minio/minio/blob/068ad33c9a1bf52b81597dcc271ab80390f9926b/pkg/event/target/queuestore.go#L160-L163

**Comment by nitisht** [comment]
Fixed by #9884

**Comment by juv** [comment]
Apparently this is not fixed by #9884 
It happened this night again for our cronjobs which ingest data to minio.
The deployment was running MinIO version `RELEASE.2020-07-02T00-15-09Z` with enabled queueDir

![image](https://user-images.githubusercontent.com/8269353/86444023-6cc1fb80-bd10-11ea-9ab9-1b79a46a7d4e.png)

**Comment by Praveenrajmani** [comment]
I see that this patch is not included in the release @juv

**Comment by harshavardhana** [comment]
> Apparently this is not fixed by #9884
> It happened this night again for our cronjobs which ingest data to minio.
> The deployment was running MinIO version `RELEASE.2020-07-02T00-15-09Z` with enabled queueDir
> 
> ![image](https://user-images.githubusercontent.com/8269353/86444023-6cc1fb80-bd10-11ea-9ab9-1b79a46a7d4e.png)

It will be fixed in another release @juv feel free to use the minio/minio:edge or master branch directly
